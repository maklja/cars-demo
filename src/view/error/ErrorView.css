.error-message {
	font-size: 3em;
	text-align: center;
	padding: 6em 0;
	background-color: #eeeeee;
}
