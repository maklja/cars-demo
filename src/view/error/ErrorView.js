import PropTypes from 'prop-types';
import React from 'react';

import './ErrorView.css';

export const ErrorView = ({ message, className, error }) => {
	return (
		<div className={className}>
			<div className="error-message">{message || error.toString()}</div>
		</div>
	);
};

ErrorView.propTypes = {
	message: PropTypes.string,
	className: PropTypes.string,
	error: PropTypes.instanceOf(Error).isRequired
};

ErrorView.defaultProps = {
	message: '',
	className: ''
};

// create HOC for error handling
export const withErrorHandler = (
	Component,
	{ message, className = '' } = {}
) => {
	return props => {
		const { error } = props;

		const view =
			error !== null ? (
				<ErrorView
					message={message}
					className={className}
					error={error}
				/>
			) : (
				<Component {...props} />
			);

		return view;
	};
};
