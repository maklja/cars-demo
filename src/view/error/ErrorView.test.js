import React from 'react';
import renderer from 'react-test-renderer';

import { ErrorView } from './ErrorView';

it('ErrorView renders correctly with error', () => {
	const error = new Error('Request error');
	const tree = renderer.create(<ErrorView error={error} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('ErrorView renders correctly with error and message prop', () => {
	const error = new Error('Request error');
	const message = 'Unable to fetch cars from server.';
	const tree = renderer
		.create(<ErrorView message={message} error={error} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('ErrorView renders correctly with className and error', () => {
	const error = new Error('Request error');
	const tree = renderer
		.create(<ErrorView className="error-msg" error={error} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
