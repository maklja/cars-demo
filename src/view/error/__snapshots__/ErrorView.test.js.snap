// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders correctly with className and error 1`] = `
<div
  className="error-msg"
>
  <div
    className="error-message"
  >
    Error: Request error
  </div>
</div>
`;

exports[`ErrorView renders correctly with error 1`] = `
<div
  className=""
>
  <div
    className="error-message"
  >
    Error: Request error
  </div>
</div>
`;

exports[`ErrorView renders correctly with error and message prop 1`] = `
<div
  className=""
>
  <div
    className="error-message"
  >
    Unable to fetch cars from server.
  </div>
</div>
`;
